src/devices_models/devices/AC_branches.jl

```diff
@@ -387,13 +387,15 @@ function add_constraints!(
         slack_lb = get_variable(container, FlowActivePowerSlackLowerBound(), T)
     end
 
+    # TODO: loop order
```

Comment:
this is interesting, and PSI uses mixed patterns.

Julia is column based, so if you vector is `array[name, t]`
you should iterate as:
```julia
for t in T
    for name in names
        # do something with array[name, t]
    end
end
```
which is equivalent to:
```julia
for t in T, name in names
    # do something with array[name, t]
end
```
In the case of sums:
```julia
sum(array[name, t] for t in T, name in names)
```
equivalent to:
```julia
sum(array[name, t] for t in T for name in names)
```
almost equivalent to:
```julia
sum(sum(array[name, t] for name in names) for t in T )
```

Reply:
so names should typically be the inner loop and t the outer loop.
but if the operation inside the loop is complex both are almost the same.
I would keep the pattern so style would lead you to the correct choices in thigh loops.

Reply:
Yeah, I originally wrote all the loops in the right order and then had to modify some depending on the context.